src/Native/Windows/build-native.cmd

```diff
@@ -10,36 +10,53 @@ set __IntermediatesDir=""
 set __BuildArch=AnyCPU
 set __VCBuildArch=x86_amd64
 set CMAKE_BUILD_TYPE=Release
+set __EnvironmentSet=false
+set __VSVersion=vs2013
+set __PlatformToolset="v120"
+
+REM Force PDB generation for all builds
+set "__LinkArgs=/DEBUG /DEBUGTYPE:cv,fixup /PDBCOMPRESS"
 
 :Arg_Loop
 :: Since the native build requires some configuration information before msbuild is called, we have to do some manual args parsing
-:: For consistency with building the managed components, args are taken in the msbuild style i.e. /p:
+:: For consistency with building the managed components, some args are taken in the msbuild style i.e. /p:
 if [%1] == [] goto :ToolsVersion
 if /i [%1] == [/p:ConfigurationGroup]    (
-    if /i [%2] == [Release] (set CMAKE_BUILD_TYPE=Release&&shift&&shift&goto Arg_Loop)
-    if /i [%2] == [Debug]   (set CMAKE_BUILD_TYPE=Debug&&shift&&shift&goto Arg_Loop)
+    if /i [%2] == [Release]         ( set CMAKE_BUILD_TYPE=Release&&shift&&shift&goto Arg_Loop)
+    if /i [%2] == [Debug]           ( set CMAKE_BUILD_TYPE=Debug&&shift&&shift&goto Arg_Loop)
     echo Error: Invalid configuration args "%1 and %2"
     exit /b 1
 )
-if /i [%1] == [/p:Platform]    (
-    if /i [%2] == [AnyCPU]  (set __BuildArch=AnyCPU&&set __VCBuildArch=x86_amd64&&shift&&shift&goto Arg_Loop)
-    if /i [%2] == [x86]     (set __BuildArch=x86&&set __VCBuildArch=x86&&shift&&shift&goto Arg_Loop)
-    if /i [%2] == [arm]     (set __BuildArch=arm&&set __VCBuildArch=x86_arm&&shift&&shift&goto Arg_Loop)
-    if /i [%2] == [x64]     (set __BuildArch=x64&&set __VCBuildArch=x86_amd64&&shift&&shift&goto Arg_Loop)
+if /i [%1] == [/p:Platform]         (
+    if /i [%2] == [AnyCPU]          ( set __BuildArch=AnyCPU&&set __VCBuildArch=x86_amd64&&shift&&shift&goto Arg_Loop)
+    if /i [%2] == [x86]             ( set __BuildArch=x86&&set __VCBuildArch=x86&&shift&&shift&goto Arg_Loop)
+    if /i [%2] == [arm]             ( set __BuildArch=arm&&set __VCBuildArch=x86_arm&&shift&&shift&goto Arg_Loop)
+    if /i [%2] == [x64]             ( set __BuildArch=x64&&set __VCBuildArch=x86_amd64&&shift&&shift&goto Arg_Loop)
+    if /i [%2] == [amd64]           ( set __BuildArch=x64&&set __VCBuildArch=x86_amd64&&shift&&shift&goto Arg_Loop)
     echo Error: Invalid platform args "%1 and %2"
     exit /b 1
 )
-if /i [%1] == [-intermediateDir]    (
-    set __IntermediatesDir=%2
-)
-if /i [%1] == [-binDir]    (
-    set __CMakeBinDir=%2
+if /i [%1] == [-intermediateDir]    ( set __IntermediatesDir=%2&&shift&&shift&goto Arg_Loop)
```

Comment:
>For these arguments I assume you cannot pass them through build.cmd because they would cause the managed build to fail.

Yes. The arguments that use a hyphen were done so because I didn't think they would be generally passed to build-native through build.cmd.

>Do you know if any of these are actually used?

The args with the hyphens are all used by the internal wrapper proj that is in my Codeflow review.

>If they are we should consider having them follow the same pattern as the other options.

We could certainly do that. I went with this format because I didn't want to potentially overwrite a property that has an existing meaning already.If I prepend "native" to the names then that worry should be gone, so I'll do that.